rsocket-core/src/main/java/io/rsocket/internal/BaseDuplexConnection.java

```diff
@@ -1,16 +1,29 @@
 package io.rsocket.internal;
 
+import io.netty.buffer.ByteBuf;
 import io.rsocket.DuplexConnection;
 import reactor.core.publisher.Mono;
 import reactor.core.publisher.MonoProcessor;
 
 public abstract class BaseDuplexConnection implements DuplexConnection {
-  private MonoProcessor<Void> onClose = MonoProcessor.create();
+  protected MonoProcessor<Void> onClose = MonoProcessor.create();
+
+  @SuppressWarnings("deprecation")
+  protected UnboundedProcessor<ByteBuf> sender = new UnboundedProcessor<>();
```

Comment:
To make `UnboundedProcessor` completely internal, this should be a private field as well. Perhaps expose a `Publisher<ByteBuf>` protected method for sub-classes to use?